### ISSUE TITLE
About link keys

### ISSUE THREAD

**Issue by fraca7** [matlo/l2cap_proxy#5]
The /var/lib/bluetooth/\<dongle addr\>/linkkeys file does not seem to work in "recent" distributions (tested on Ubuntu 20 and Raspberry Pi OS). Instead, link keys have to be stored in an ini-like file in a subdirectory (at least that's what works for me on said distros). The file is named "info" and resides in /var/lib/bluetooth/\<dongle addr\>/\<device addr\>/. It has the following format:

```
[LinkKey]
Key=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
Type=4
PINLength=0
```

I think this behavior dates back to Ubuntu 14 but I'm not sure.

**Comment by fraca7** [comment]
Update: since I discovered hci_write_stored_link_key I was going to add command-line arguments to specify the link keys directly, then I noticed bt_utils.c has a write_stored_link_key function, which is unused. Is there a reason for this ?

**Comment by matlo** [comment]
I probably used it at some point, while trying to make it work for the PS4/DS4. There is no guarantee this function works and/or that it is sufficient to have link encryption correctly configured on the bluetooth chip.

**Comment by fraca7** [comment]
Indeed, BlueZ uses BT management sockets (https://github.com/bluez/bluez/blob/master/doc/mgmt-api.txt) to communicate "directly" with the kernel. I'll go with that then. The link key files are too distribution-dependent (the ones I mentioned are actually parsed by gnome-bluetooth it seems)

**Comment by matlo** [comment]
You may be interested by: https://github.com/matlo/GIMX/blob/master/core/connectors/bluetooth/linux/bt_mgmt.c

**Comment by fraca7** [comment]
Indeed, thanks
